File: LeetCode/Easy/0448-find-all-numbers-disappeared-in-an-array/0448-find-all-numbers-disappeared-in-an-array.java
class Solution {
    public List<Integer> findDisappearedNumbers(int[] nums) {
        int[] checker = new int[nums.length + 1];

        for (int n : nums) {
            checker[n]++;
        }

        List<Integer> ans = new ArrayList<>();

        for (int i = 1; i < checker.length; ++i) {
            if(checker[i] == 0) ans.add(i);
        }

        return ans;
    }
}

File: LeetCode/Easy/0448-find-all-numbers-disappeared-in-an-array/0448-find-all-numbers-disappeared-in-an-array.kt
class Solution {
    fun findDisappearedNumbers(nums: IntArray): List<Int> {

        val checker = IntArray(nums.size+1) {0}
        val ans: MutableList<Int> = mutableListOf<Int>()

        for (n in nums) {
            checker[n]++
        }

        for ((idx, check) in checker.withIndex()) {
            if (idx == 0) continue
            if (check == 0) ans.add(idx)
        }

        return ans
    }
}